aspnetcore/security/authorization/secure-data.md

```diff
@@ -0,0 +1,314 @@
+
+## Inject the authorization service into the views
+
+Currently the UI shows edit and delete links for data the user cannot modify. We'll fix that by applying the authorization handler to the views.
+
+Inject the authorization service in the *Views/_ViewImports.cshtml* file so it will be available to all views:
+
+[!code-html[Main](secure-data/samples/final/Views/_ViewImports.cshtml)]
+
+Update the *Views/Contacts/Index.cshtml* Razor view to show only display the edit and delete links for users who can edit/delete the contact.
+
+Add `@using ContactManager.Authorization;`
+
+Update the `Edit` and `Delete` links so they are only rendered for users with permission to edit and delete the contact.
+
+[!code-html[Main](secure-data/samples/final/Views/Contacts/Index.cshtml?range=63-84)]
+
+Warning: Hiding links from users that do not have permission to edit or delete data does not secure that app, it makes the app more user friendly by displaying only valid links. Users can hack the generated URLs to invoke edit and delete operations on data they don't own.  The controller must repeat the access checks to be secure.
+
+### Update the Details view
+
+Update the details view so managers can approve or reject contacts:
+
+[!code-html[Main](secure-data/samples/final/Views/Contacts/Details.cshtml?range=53-)]
+
+## Test the completed app
+
+If you have run the app and have contacts, delete all the records in the `Contact` table and restart the app to seed the database. You need to exit and restart IIS Express to seed the database.
+
+If you deleted the database, run update on the database:
+
+```none
+   dotnet ef database update
+```
+
+You'll need to register a user to browse the contacts.
+
+An easy way to test the completed app is to launch three different browsers (or incognito/InPrivate versions). In one browser, register a new user, for example, `test@example.com`. SignIn to each browser with a different user. Verify the following:
+
+* Registered users can view all the approved contact data.
+* Registered users can edit/delete their own data. 
+* Managers can approve or reject contact data. The `Details` view shows **Approve** and **Reject** buttons. 
+* Administrators can approve/reject and edit/delete any data.
+
+| User| Options |
+| ------------ | ---------|
+| admin@contoso.com | Can edit/delete and approve/reject all data|
+| manager@contoso.com | Can approve/reject and edit/delete own data  |
+| test@example.com | Can edit/delete own data |
+
+Create a contact in the administrators browser. Copy the URL for delete and edit from the administrator contact. Paste these links into the test users browser to verify the test user cannot perform these operations.
+
+## Create the starter app
+
+Follow these instructions to create the starter app.
+
+* Create a new **ASP.NET Core Web Application** using [Visual Studio 2017](https://www.visualstudio.com/en-us/visual-studio-homepage-vs.aspx) named "ContactManager"
```

Comment:
That's just to create the starter app from scratch, which is optional. VS Code folks will probably download the starter.

BTW, I really hate writing "to create it from scratch" but I can't think of a better way to say that.